docs/redis.md

```diff
@@ -26,6 +35,10 @@ metrics:
       address: localhost:6379
       collection_interval: 30s
       password: pwd
+      insecure: false
```

Comment:
Can we add some unit tests for setting TLS?

* A `valid` test for `secure` path
* A `valid` test for the `insecure` path
* Some `invalid` tests to verify the errors we throw when conflicting parameter values are specified together (e.g. `insecure` is true or omitted, but `cert_file` is set

Reply:
@qingling128 How would we throw an error on your 3rd request? The only kind of config validation that I'm aware of in the ops agent right now are things like data type validation (is this a valid URL), all the deeper validation is currently in the receivers themselves, which don't show until config has been generated & otel collector attempts to run

Reply:
https://github.com/go-playground/validator#other has a few options that could work for the case here: e.g. `required_without` and `exclude_with`.

Something like:

```
  A string `validate:"required_without=B,excluded_with=B"`
```

Reply:
setup that validation and those test cases

Reply:
Without an explicit config option for TLS, there's now no way to enable TLS without setting explicit certificates. I don't know enough about how Redis is typically deployed; is it really the case that Redis is never deployed with normal globally-trusted SSL certificates? (in which case the host defafult SSL